File: YouMedServer/Models/Entities/Doctor.cs
using System.ComponentModel.DataAnnotations;

namespace YouMedServer.Models.Entities
{
    public class Doctor
    {
        [Key]
        public int DoctorID { get; set; }

        public string? Introduction { get; set; }

        public int? Experience { get; set; }

        public int UserID { get; set; }

        public required User User { get; set; }

        public int ClinicID { get; set; }

        public Clinic? Clinic { get; set; }
        public bool IsDeleted { get; set; } = false;
    }
}

File: YouMedApp/types/Doctor.ts
import Clinic from "./Clinic";
import Specialties from "./Specialties";
import User from "./User";

export default interface Doctor {
  doctorID: number;
  introduction: string; 
  experience: number; 
  user: User;
  clinicID: number;
  clinic: Clinic;
  specialties?: Specialties[]; 
  createdAt: string;
}
